OpenSubtitlesHandler/Util.cs

```diff
@@ -0,0 +1,152 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Text;
+using System.Text.Json;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace OpenSubtitlesHandler
+{
+    public static class Util
+    {
+        private static readonly HttpClient HttpClient = new HttpClient();
+        private static readonly JsonSerializerOptions SerializerOpts = new JsonSerializerOptions { IncludeFields = true, PropertyNamingPolicy = SnakeCaseNamingPolicy.Instance };
+        public static Action<string> OnHttpUpdate = _ => {};
+        private static string _version = string.Empty;
+
+        public static DateTime NextReset
+        {
+            get
+            {
+                // download limits get reset every day at midnight (UTC)
+                var now = DateTime.UtcNow;
+
+                return new DateTime(now.Year, now.Month, now.Day).AddDays(1).AddMinutes(1);
+            }
+        }
+
+        internal static void SetVersion(string version)
```

Comment:
This doesn't seem thread safe.

Reply:
Could you elaborate? (It's called only once at the startup of the plugin, in the constructor)